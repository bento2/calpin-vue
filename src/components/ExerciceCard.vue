<script setup lang="ts">
import type { Exercice } from '@/types/ExerciceSchema.ts'
import ExerciceIcon from '@/components/ExerciceIcon.vue'

const props = defineProps<{ exercice: Exercice }>()
const { exercice } = props
</script>

<template>
  <v-card class="" color="indigo" elevation="2" outline>
    <v-card-text class="d-flex align-center font-bold mb-1 ga-1">
      <ExerciceIcon :exercice="exercice" />
      <div class="font-bold flex-grow-1">{{ exercice.name }}</div>
      <slot name="actions"></slot>
    </v-card-text>
  </v-card>
</template>

<style scoped></style>
