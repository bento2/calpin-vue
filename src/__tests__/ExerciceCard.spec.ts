import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'

import ExerciceCard from '@/components/ExerciceCard.vue'
import type { Exercice } from '@/types/ExerciceSchema.ts'

const exerciceMock: Exercice = {
  id: '1',
  name: 'Pompes',
  icon: '/icons/push-up.png',
  difficulty: 'dÃ©butant',
  hasEquipment: true
}

describe('ExerciceCard', () => {
  it('Render ExerciceCard', () => {
    const wrapper = mount(ExerciceCard, {
      props: {
        exercice: exerciceMock
      }
    })
    expect(wrapper.text()).toContain(exerciceMock.name)
  })

  it('Check snapshot', () => {
    const wrapper = mount(ExerciceCard, {
      props: {
        exercice: exerciceMock
      },
      global: { stubs: { 'n-image': true } }
    })

    // Snapshot test
    expect(wrapper.html()).toMatchSnapshot()
  })
})
