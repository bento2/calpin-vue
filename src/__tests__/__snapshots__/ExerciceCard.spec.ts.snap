// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExerciceCard > Check snapshot 1`] = `
"<div class="v-card v-theme--light v-card--density-default elevation-2 v-card--variant-elevated bg-blue-accent-2" outline="">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <div class="v-progress-linear__indeterminate">
        <div class="v-progress-linear__indeterminate long"></div>
        <div class="v-progress-linear__indeterminate short"></div>
      </div>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class="v-card-text d-flex align-center font-bold mb-1 ga-1">
    <div data-v-18f349dd="" class="v-avatar v-theme--light bg-transparent v-avatar--density-default v-avatar--variant-flat border" style="width: 56px; height: 56px; border: 2px solid black;">
      <div data-v-18f349dd="" class="v-responsive v-img v-img--booting" aria-label="Pompes" role="img">
        <div class="v-responsive__sizer"></div><img class="v-img__img v-img__img--cover fade-transition-enter-from fade-transition-enter-active" src="/icons/push-up.png" alt="Pompes" style="display: none;">
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
      <!----><span class="v-avatar__underlay"></span>
    </div>
    <div class="font-bold flex-grow-1">Pompes</div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
